netplugin/agent/docker_event.go

```diff
@@ -94,7 +95,69 @@ func handleDockerEvents(event *dockerclient.Event, ec chan error, args ...interf
 		if err != nil {
 			log.Errorf("Event:'die' Http error posting endpoint update, Error:%s", err)
 		}
-	}
+
+	case "":
+
+		/* Handles the case when we get a docker event with null event status
+		   such as when we proactively create containers ourselves rather than the end user
+		   Look for the "connect" action in such cases.
+		*/
+
+		containerID := event.Actor.Attributes["container"]
+		networkID := event.Actor.ID
+
+		if event.Action == "connect" && containerID != "" {
+
+			endpointUpdReq.Event = "connect"
+			defaultHeaders := map[string]string{"User-Agent": "engine-api-cli-1.0"}
+			cli, err := client.NewClient("unix:///var/run/docker.sock", "v1.21", nil, defaultHeaders)
+			if err != nil {
+				panic(err)
```

Comment:
Any Reason to panic if we cant reach the docker client ? I would think logging an error may be the right way to go about it instead of exiting netplugin.